### ISSUE TITLE
Make sorting smarter

### ISSUE THREAD

**Issue by wilsonpage** [mozilla-magnet/magnet-client#98]
The scroll-view list items are not sorted intelligently. We should define a sorting 'score' based on when the tile was last seen and how close the beacon is to the user.

Rule: Closest most recently seen tile wins.

**Comment by wilsonpage** [comment]
@arcturus @samgiles also need your input on this one 😄

**Comment by wilsonpage** [comment]
To simplify things we could just sort by distance. If a beacon hasn't been seen for `n` seconds then we can no longer be sure of its distance, so its distance becomes `undefined`, moving it further down the list.

**Comment by isabelrios** [comment]
Having the history and the distance (almost ready to be shown) have you already agreed a way of sorting the tiles @wilsonpage? 
Now that we notify about new web pages around I think that it is important that those new pages are shown at the beginning of the list or at least easy to identify to know which ones are the new ones. Since we have the whole list of tiles discovered, including the ones that are not in rage, it is not easy and a bit confusing to know which urls are the ones discovered when going to the app from a notification.

Also, I think this would be good to have to avoid the fact that when you refresh the app being in the same conditions (same beacons, same place), the tiles are sorted in different ways. That could be very confusing when using the app by new users.

**Comment by wilsonpage** [comment]
I really like the idea of visually marking 'new' (not previously discovered) tiles in some way. Like when you open your inbox, you can quickly see which items require your attention.

**Comment by isabelrios** [comment]
That would be a nice solution :) 
Ccing @lime124, Tif maybe you can give us your view con this

**Comment by lime124** [comment]
Yes that would be useful too. I also made a comment just today (as I catch up on old email) that it might be worth thinking about how does content expire - you've seen it x number of times and you haven't interacted with it. Perhaps you're just not interested. I think  all of this is going to require numerous iterations and trials with people to get it right.
